### PR TITLE
Switch bootstrap to use entry_points, cleanup packaging a little

### PR DESCRIPTION
**Pull request by cmaloney**
Should help with some of the dirty build problems we've been seeing

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -82,5 +82,5 @@ deps=
   teamcity-messages
 changedir=packages/bootstrap/extra
 commands=
-  python setup.py install
+  pip install .
```
Comment (orsenthil) [inline tox.ini]: To maintain consistency with `packages/bootstrap/build` using `pip3` might be a good idea.

**Comment by spahl** [comment]
:ship:

File: packages/bootstrap/extra/dcos_internal_utils/bootstrap/cli.py [hunk 2]
```diff
@@ -84,7 +84,3 @@ def parse_args():
         default='/opt/mesosphere/etc/master_count',
         help='File with number of master servers')
     return parser.parse_args()
-
-
-if __name__ == '__main__':
-    main()
```
Comment (cmaloney) [inline packages/bootstrap/extra/dcos_internal_utils/bootstrap/cli.py]: nit: should remove the shebang at the top as well, but needing to land this quickly as a _lot_ of builds are borked currently.

File: tox.ini (hunk 1 above)
Comment (cmaloney) [inline tox.ini]: here we should be running inside of a python 3.4 virtual environment, so _should_ be safe. That said, be a nice after this lands misc. cleanup PR (Trying to land this quick to un-break)

Comment (cmaloney) [inline tox.ini]: Should actually replace all `pip` usage in our codebase with `pip3` (This is far from the only one)

**Pull request merged**